qiskit/aqua/operators/list_ops/list_op.py

```diff
@@ -16,6 +16,7 @@
 
 from functools import reduce
 from typing import List, Union, Optional, Callable, Iterator, Set, Dict
+from numbers import Number
```

Comment:
This is quite a heavy import, we had this before for the quantum register but had to remove it again due to the slowdown. There, we just explicitly put all number types as something like ``isinstance(number, (int, np.int, np.int64, ...))`` could we do the same here?

Reply:
My recollection is that the import of the  `Number` module itself was not so heavy in terms of LOC. But, M Treinish did some benchmarks that showed a significant performance hit. The variable in question was used as an index. If the check in the present case occurs once among  much more expensive computations, then it might be worth it. On the other hand if a more narrow, specific check can be done, I agree it's better. If it needs something broader, like `Number` it can be changed later.

Reply:
These are the times of doing that instance check with Number versus a tuple(complex, float, int)
```
2.724337100982666e-06
1.343618631362915e-06
```
over one million calls in a loop where the overall time was divided by the loop count. First is with Number, second in the tuple. 

I had started with a tuple of different types but numpy had so many, this seemed "safer" since it really depends on the combo fn result. In general there is one test here for each listop in the operator. Given all the other matrix conversions etc that are going on I doubt this is really much of a performance issue - certainly given the  quick test as above does not seem it should be.